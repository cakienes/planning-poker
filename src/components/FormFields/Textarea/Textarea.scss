.defaultTextareaHolder {
    margin-top: 20px;
    .defaultTextarea {
        width: 100%;
        font-size: 24px;
        padding: 20px 10px;
        font-size: 24px;
    }

    .defaultLabel {
        padding: 0 10px;
        font-size: 22px;
    }

    .error {
        border: 1px solid red;
    }
}
