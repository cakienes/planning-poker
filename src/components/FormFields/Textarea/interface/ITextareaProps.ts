import { WrappedFieldProps } from 'redux-form';

export default interface ITextareaProps extends WrappedFieldProps {
    label?: string;
    placeholder?: string;
}
