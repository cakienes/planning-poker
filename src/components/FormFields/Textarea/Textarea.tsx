import React from 'react';
import ITextareaProps from './interface/ITextareaProps';
import './Textarea.scss';

const Textarea: React.FC<ITextareaProps> = ({ label, input, meta: { touched, error }, placeholder }) => {
    return (
        <div className="defaultTextareaHolder">
            <label className="defaultLabel">{label}</label>
            <div>
                <textarea className="defaultTextarea" id={input.name} placeholder={placeholder} rows={10} {...input} />
            </div>
        </div>
    );
};

export default Textarea;
