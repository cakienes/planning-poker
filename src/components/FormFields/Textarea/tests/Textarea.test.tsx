import { shallow } from 'enzyme';
import React from 'react';
import ITextareaProps from '../interface/ITextareaProps';
import Textarea from '../Textarea';

describe('Textarea.tsx', () => {
    let textareaProps: ITextareaProps = {
        label: 'label',
        placeholder: 'placeholder',
        input: {
            name: 'name',
            onBlur: () => {},
            onChange: () => {},
            onDragStart: () => {},
            onDrop: () => {},
            onFocus: () => {},
            value: null,
        },
        meta: {
            asyncValidating: false,
            autofilled: false,
            dirty: false,
            // @ts-ignore
            dispatch: null,
            form: 'form',
            initial: false,
            invalid: false,
            pristine: false,
            submitFailed: false,
            submitting: false,
            touched: false,
            valid: false,
            visited: false,
        },
    };
    it('renders correctly TextArea', async () => {
        expect(shallow(<Textarea {...textareaProps} />)).toMatchSnapshot();
    });

    it('renders without crashing while error', async () => {
        textareaProps = {
            ...textareaProps,
            meta: {
                ...textareaProps.meta,
                error: true,
                touched: true,
            },
        };
        expect(shallow(<Textarea {...textareaProps} />)).toMatchSnapshot();
    });
});
