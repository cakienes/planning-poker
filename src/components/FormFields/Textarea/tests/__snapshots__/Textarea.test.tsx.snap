// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea.tsx renders correctly TextArea 1`] = `ShallowWrapper {}`;

exports[`Textarea.tsx renders without crashing while error 1`] = `ShallowWrapper {}`;
